google/devtools/clouddebugger/v2/clouddebugger_gapic.yaml

```diff
@@ -0,0 +1,167 @@
+type: com.google.api.codegen.ConfigProto
+language_settings:
+  java:
+    package_name: com.google.cloud.devtools.clouddebugger.spi.v2
+  python:
+    package_name: google.cloud.devtools.clouddebugger.v2
+  go:
+    package_name: cloud.google.com/go/devtools/clouddebugger/apiv2
+  csharp:
+    package_name: Google.Devtools.Clouddebugger.V2
+  ruby:
+    package_name: Google::Cloud::Devtools::Clouddebugger::V2
+  php:
+    package_name: Google\Cloud\Devtools\Clouddebugger\V2
+  nodejs:
+    package_name: '@google-cloud/clouddebugger'
+interfaces:
+- name: google.devtools.clouddebugger.v2.Debugger2
+  collections: []
+  retry_codes_def:
+  - name: idempotent
+    retry_codes:
+    - UNAVAILABLE
+    - DEADLINE_EXCEEDED
+  - name: non_idempotent
+    retry_codes: []
+  retry_params_def:
+  - name: default
+    initial_retry_delay_millis: 100
+    retry_delay_multiplier: 1.3
+    max_retry_delay_millis: 60000
+    initial_rpc_timeout_millis: 60000
+    rpc_timeout_multiplier: 1
+    max_rpc_timeout_millis: 60000
+    total_timeout_millis: 600000
+  methods:
+  - name: SetBreakpoint
+    flattening:
+      groups:
+      - parameters:
+        - debuggee_id
+        - breakpoint
+        - client_version
+    required_fields:
+    - debuggee_id
+    - breakpoint
+    - client_version
+    request_object_method: true
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    timeout_millis: 60000
+  - name: GetBreakpoint
+    flattening:
+      groups:
+      - parameters:
+        - debuggee_id
+        - breakpoint_id
+        - client_version
+    required_fields:
+    - debuggee_id
+    - breakpoint_id
+    - client_version
+    request_object_method: true
+    retry_codes_name: idempotent
+    retry_params_name: default
+    timeout_millis: 60000
+  - name: DeleteBreakpoint
+    flattening:
+      groups:
+      - parameters:
+        - debuggee_id
+        - breakpoint_id
+        - client_version
+    required_fields:
+    - debuggee_id
+    - breakpoint_id
+    - client_version
+    request_object_method: true
+    retry_codes_name: idempotent
+    retry_params_name: default
+    timeout_millis: 60000
+  - name: ListBreakpoints
+    required_fields:
+    - debuggee_id
+    - include_all_users
```

Comment:
Ok, "required === default value is an error" makes more sense. Is that actually what the generated library enforces? E.g. the user can't pass `0` for required numbers, `""` for required strings, and `[]` for required bytes?

It still shouldn't ever happen for boolean fields like this one :) It would mean "this field needs to be set to true", which doesn't make sense.

From inspecting the generated config - the tool is just assuming all fields of the request are required?